src/core/biblio.js

```diff
@@ -137,19 +95,78 @@ export async function run(conf) {
     idbRefs.push(...neededRefs.map(id => ({ id, data: null })));
     console.warn(err);
   }
-  const split = { hasData: [], noData: [] };
-  idbRefs.forEach(ref => {
-    (ref.data ? split.hasData : split.noData).push(ref);
-  });
-  split.hasData.forEach(ref => {
-    biblio[ref.id] = ref.data;
-  });
-  const externalRefs = split.noData.map(item => item.id);
-  if (externalRefs.length) {
-    // Going to the network for refs we don't have
-    const data = await updateFromNetwork(externalRefs, { forceUpdate: true });
-    Object.assign(biblio, data);
+
+  return idbRefs;
+}
+
+export default class {
```

Comment:
Maybe we could add a proper name and check that instead. `class` is a keyword so that can't be used, do you have a good idea?

Reply:
`klass` is pretty common

Reply:
Decided to use `Class` as using uppercase matches our code style 👀

Reply:
Smart

Reply:
Wait. How about `Plugin`?

Reply:
Every modules are Plugin. I'm not sure they are really plugins, though. (No real plugin API, need to modify profile whenever we need third party modules.)

Reply:
Although @saschanaz is technically correct about not really being "plugins", we've always referred to these as plugins. The Class thing looks a bit weird, tbh... so I'd prefer we just call them `Plugin` for legacy reasons and hold our noses. Naming things is hard.

Reply:
Okay, reasonable. Actually I like the Plugin name too, maybe one day we could make them real plugins...

Reply:
I would love that :) The ultimate plugins would all be their own git repos and we'd have a nicer plugin API. What's some nice plugins APIs you've seen?

Reply:
Not sure about what's "nice", but I would mention the plugins we have in our `package.json` and the APIs they are using (from Babel, ESLint etc.).